### PR TITLE
Dispose FSharpProjectPackage properly

### PR DESCRIPTION
**Pull request by dungpa**
This is supposed to be part of https://github.com/Microsoft/visualfsharp/pull/2043.
Sorry for missing this earlier.

### CODE REVIEW

File: vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs [hunk 1]
```diff
@@ -355,6 +355,14 @@ namespace rec Microsoft.VisualStudio.FSharp.ProjectSystem
                 pIdIco <- 400u
                 VSConstants.S_OK
 
+            override this.Dispose(disposing) =
```
Comment (KevinRansom) [inline vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs]: Will this always happen on a UI thread? should we used interlocked compex on componentID to ensure it's not double disposed, or is double disposing Okay?

Comment (KevinRansom) [inline vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs]: I guess I mean double revoking.

Comment (dungpa) [inline vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs]: We guard revoking component by `componentId` and `mgr` so I think double revoking should not happen.

Comment (KevinRansom) [inline vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs]: It doesn't look thread safe.  But there may be other mitigations in place.  

However, as it stands two threads could both invoke mgr..FRevokeComponent(componentID) simultaneously.

Now that might not be a problem ... 
  for example if mgr..FRevokeComponent(componentID) is resilient to double revoke.

or
 Something higher up the stack is ensuring that two threads can't dispose simultaneous ... for example something enforces that only a single thread at a time is doing the dispose.

to make it threadsafe either grabbing a lock and double checking or using Interlocked.CompareExchange to set componentID back to zero would probably work fine.

Comment (KevinRansom) [inline vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs]: Looks good to me.

Thank you.

Kevin

**Pull request merged**